barman/cli.py

```diff
@@ -918,47 +928,58 @@ def recover(args):
+    # If the backup to be recovered is compressed then there are additional
+    # checks to be carried out
+    if backup_id.compression is not None:
         # Set the recovery staging path from the cli if it is set
         if args.recovery_staging_path is not None:
             try:
-                recovery_staging_path = parse_recovery_staging_path(
-                    args.recovery_staging_path
-                )
+                recovery_staging_path = parse_staging_path(args.recovery_staging_path)
             except ValueError as exc:
                 output.error("Cannot parse recovery staging path: %s", str(exc))
                 output.close_and_exit()
             server.config.recovery_staging_path = recovery_staging_path
-
-        # If no staging path is found then we need to display a proper error
-        # according to the backup type (compressed or incremental)
+        # If the backup is compressed but there is no recovery_staging_path
+        # then this is an error - the user *must* tell barman where recovery
+        # data can be staged.
         if server.config.recovery_staging_path is None:
-            if backup_id.parent_backup_id is not None:
-                output.error(
-                    "Cannot recover from backup '%s' of server '%s': "
-                    "backup will be combined with pg_combinebackup in the "
-                    "barman host but no recovery staging path is provided. "
-                    "Either set recovery_staging_path in the Barman config "
-                    "or use the --recovery-staging-path argument.",
-                    args.backup_id,
-                    server.config.name,
-                )
-                output.close_and_exit()
+            output.error(
+                "Cannot recover from backup '%s' of server '%s': "
+                "backup is compressed with %s compression but no recovery "
+                "staging path is provided. Either set recovery_staging_path "
+                "in the Barman config or use the --recovery-staging-path "
+                "argument.",
+                args.backup_id,
+                server.config.name,
+                backup_id.compression,
+            )
+            output.close_and_exit()
 
-            if backup_id.compression is not None:
-                output.error(
-                    "Cannot recover from backup '%s' of server '%s': "
-                    "backup is compressed with %s compression but no recovery "
-                    "staging path is provided. Either set recovery_staging_path "
-                    "in the Barman config or use the --recovery-staging-path "
-                    "argument.",
-                    args.backup_id,
-                    server.config.name,
-                    backup_id.compression,
-                )
+    # If the backup to be recovered is incremental then there are additional
+    # checks to be carried out
+    if backup_id.parent_backup_id is not None:
```

Comment:
```suggestion
    if backup_id.is_incremental:
```

We now have a property :)

Reply:
We will address this through a separate PR as there are more occurrences of this same situation in other pieces of the code.